src/main/java/fr/ecolnum/projectapi/DTO/CandidateDto.java

```diff
@@ -51,12 +52,22 @@ public CandidateDto(Candidate candidate) {
         }
     }
 
+    public Candidate convertToCandidateObject(final GroupRepository groupRepository) throws IdNotFoundException {
+        Set<Group> belongsTo = extractSetFromRepository(groupRepository, this.belongsToGroup);
+
+        return new Candidate(this.id, this.firstName, this.lastName, belongsTo);
+    }
```

Comment:
why create a new function? the Candidate model doesn't have a set of associated Pools (Set<Pool>)